rfc/rfc017-distributed-network-grpc-v2.md

```diff
@@ -45,4 +54,270 @@ As specified by RFC015, the node MUST authenticate the peer's node DID as follow
+The individual messages describe the requirements.
+
+## 5. Gossip protocol
+
+The most efficient protocol for synchronizing transactions over nodes is to simply send new transactions to all nodes.
+This only works for the case where all the nodes are connected and when the nodes are always online.
+This might be the case for 99% of the time. For the last 1%, the set reconciliation protocol of chapter 6 is used.
+Although the set reconciliation protocol can also synchronize any difference in transactions sets between nodes, it's not very efficient for small differences.
+
+### 5.1 Data requirements
+
+#### 5.1.1 XOR
+
+The exclusive-or (XOR) value is calculated over all transaction references within a range.
+The size of the XOR value is therefore the same as the transaction reference size (SHA256: 32 bytes).
+The XOR value is used to quickly determine if two nodes have processed the exact same set of transactions.
+
+### 5.2 Operation
+
+All messages and values mentioned in this chapter are scoped to a single connection between two nodes.
+*Alice* and *Bob* are used to represent two nodes connected to each other.
+
+The protocol generally operates as follows:
+
+1. Alice sends at a set interval (§5.2.1):
+   * XOR of all known transaction references.
+   * The highest LC value.
+   * A list of transaction references since the previous message. If no previous message has been send, the list is empty.
+
+2. When receiving Alice's message, Bob compares the XOR value from Alice with its own (§5.2.2):
+   * When the XOR is the same: no action required.
+   * When different:
+      * Remove all known transaction references from the list.
+      * Request the remaining transactions.
+      * If the remaining list is empty and the LC value is equal or higher than Bob's highest LC value, Bob sends a `State` message as defined in Chapter 6. 
+
+3. Alice responds with the requested transactions (§5.2.3).
+
+4. After adding Alice's transactions to the DAG (making sure its cryptographic signature is valid), Bob SHOULD query the payload if it's missing.
+
+A node MUST make sure to only add transactions of which all previous transactions are present. If previous transactions are missing, Bob will send a `State` message.
+
+#### 5.2.1 Broadcasting new transactions
+
+A node's new transaction references MUST be broadcast at an interval using the `Gossip` message, by default every 2 seconds. The interval MAY be adjusted by the node operator but MUST conform to the limits (min/max interval) defined by the network. It is advised to keep it relatively short because it directly influences the speed by which new transactions are retrieved.
+
+The `Gossip` message MUST contain an `XOR` value, an `LC` value and a list of transaction references (`transactions`).
+The list MUST contain transaction references added to the DAG since the last `Gossip` message.
+The list of transaction references MUST be tracked per connection.
```

Comment:
What is the purpose of this requirement? It sounds like an implementation detail.
```suggestion
```

Reply:
To make clear that you have to do this per connection. If all nodes do not do this, the gossip part will fail. We could change it to SHOULD to make it less strict.